src/dcap_provider.cpp

```diff
@@ -203,34 +205,44 @@ static inline quote3_error_t fill_qpl_string_buffer(
 //
 // Determine time cache should invalidate for given collateral
 //
-bool get_cache_expiration_time(CollateralTypes collateral_type, time_t &expiration_time)
+bool get_cache_expiration_time(const string &cache_control, const string &url, time_t &expiration_time)
 {
     time_t max_age = 0;
     tm* max_age_s = localtime(&max_age);
-
-    switch(collateral_type)
+    string match = "max-age=";

```

Comment:
Why is "match" a local variable? Why not make it either a const static or a string literal with an associated length? Otherwise you're taking a round trip through the allocator on every call for a string constant.

Reply:
Fixed - made it a const static and moved with the others at top of the file